core/include/webview/detail/backends/cocoa_webkit.hh

```diff
@@ -498,35 +469,7 @@ private:
     m_window = nullptr;
     dispatch([this] { on_window_destroyed(); });
   }
-  void set_up_window() {
-    objc::autoreleasepool arp;
-
-    // Main window
-    if (m_owns_window) {
-      m_window = objc::msg_send<id>("NSWindow"_cls, "alloc"_sel);
-      auto style = NSWindowStyleMaskTitled;
-      m_window = objc::msg_send<id>(
-          m_window, "initWithContentRect:styleMask:backing:defer:"_sel,
-          CGRectMake(0, 0, 0, 0), style, NSBackingStoreBuffered, NO);
-
-      m_window_delegate = create_window_delegate();
-      objc_setAssociatedObject(m_window_delegate, "webview", (id)this,
-                               OBJC_ASSOCIATION_ASSIGN);
-      objc::msg_send<void>(m_window, "setDelegate:"_sel, m_window_delegate);
-
-      on_window_created();
-    }
-
-    set_up_web_view();
-    set_up_widget();
-
-    objc::msg_send<void>(m_window, "setContentView:"_sel, m_widget);
-
-    if (m_owns_window) {
-      objc::msg_send<void>(m_window, "makeKeyAndOrderFront:"_sel, nullptr);
-    }
-  }
-  void set_up_web_view() {
+  void m_window_settings(bool debug) {
```

Comment:
For now we just pass in `debug`, but as the code evolves this parameter scope can broaden.